crates/uniplate/src/uniplate.rs

```diff
@@ -1,12 +1,34 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum UniplateError {
+    #[error("Could not reconstruct node because some children were not given")]
+    NotEnoughChildren,
```

Comment:
(Opt) Might be nice to give an expected number vs given number here?

Reply:
Done!